dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityLiveMode.java

```diff
@@ -0,0 +1,200 @@
+
+            // Verify and handle the case for unauthorized access of this contentlet
+            Boolean unauthorized = CMSUrlUtil.getInstance().isUnauthorizedAndHandleError(htmlPage, uri, user, request, response);
+            if (unauthorized) {
+                return;
+            }
+
+            // Fire the page rules until we know we have permission.
+            RulesEngine.fireRules(request, response, htmlPage, Rule.FireOn.EVERY_PAGE);
+
+            Logger.debug(this.getClass(), "Recording the ClickStream");
+            if (Config.getBooleanProperty("ENABLE_CLICKSTREAM_TRACKING", false)) {
+                if (user != null) {
+                    UserProxy userProxy = com.dotmarketing.business.APILocator.getUserProxyAPI().getUserProxy(user,
+                            APILocator.getUserAPI().getSystemUser(), false);
+                    if (!userProxy.isNoclicktracking()) {
+                        ClickstreamFactory.addRequest((HttpServletRequest) request, ((HttpServletResponse) response), host);
+                    }
+                } else {
+                    ClickstreamFactory.addRequest((HttpServletRequest) request, ((HttpServletResponse) response), host);
+                }
+            }
+
+            // Begin page caching
+            String userId = (user != null) ? user.getUserId() : APILocator.getUserAPI().getAnonymousUser().getUserId();
+            String language = String.valueOf(langId);
+            String urlMap = (String) request.getAttribute(WebKeys.WIKI_CONTENTLET_INODE);
+            String queryString = request.getQueryString();
+            String persona = null;
+            Optional<Visitor> v = visitorAPI.getVisitor(request, false);
+            if (v.isPresent() && v.get().getPersona() != null) {
+                persona = v.get().getPersona().getKeyTag();
+            }
+
+
+            PageCacheParameters cacheParameters =
+                    new BlockPageCache.PageCacheParameters(userId, language, urlMap, queryString, persona);
+
+            boolean buildCache = false;
+            String key = VelocityUtil.getPageCacheKey(request, response);
+            if (key != null) {
+                String cachedPage = CacheLocator.getBlockPageCache().get(htmlPage, cacheParameters);
+                if (cachedPage == null || "refresh".equals(request.getParameter("dotcache"))
+                        || "refresh".equals(request.getAttribute("dotcache")) || (request.getSession(false) != null
+                                && "refresh".equals(request.getSession(true).getAttribute("dotcache")))) {
+                    // build cached response
+                    buildCache = true;
+                } else {
+                    // have cached response and are not refreshing, send it
+                    response.getWriter().write(cachedPage);
+                    return;
+                }
+            }
+
+            Writer tmpOut = (buildCache) ? new StringWriter(4096) : new VelocityFilterWriter(out);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 4096 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS109)